src/main/java/com/houarizegai/calculator/app.java

```diff
@@ -0,0 +1,11 @@
+class App {
+    static void stringEqualityCheckNoncompliant(String string1, String string2) {
+    // Noncompliant: the == operator doesn't compare the contents of the strings.
+    if(string1 == string2) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.